src/groupByKey.ts

```diff
@@ -2,6 +2,22 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T extends
+{ [key: string]: any }>(items: T[], key: string): GroupsMap<T> {
+  if (!items.length) {
+    return {};
+  }
+
+  const groupedObject: GroupsMap<T> = items
+    .reduce((acc: GroupsMap<T>, item: T) => {
+      if (acc[item[key]]) {
```

Comment:
Good job, I would just recommend using more reliable ways to check the presence of the property in object